cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAvailableNodePortAddresses(t *testing.T) {
+	testCases := []struct {
+		name                        string
+		nodePortAddressesFromConfig []string
+		excludeDevices              []string
+	}{
+		{
+			name:                        "empty nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+		{
+			name:                        "good path with nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{"1.2.3.0/24", "1.2.3.4/32"},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, _, err := getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, tc.excludeDevices)
+			assert.NoError(t, err)
```

Comment:
Unit test's purpose is ensuring the function works as expected when it's first written and when it's updated.
Unit test like this one would barely catch any mistake when the function is updated, for example, if someone updates the function to below, the unit test still show everything is fine:
```
func getAvailableNodePortAddresses(nodePortAddressesFromConfig []string, excludeDevices []string) ([]net.IP, []net.IP, error) {
    return nil, nil, nil
}
```
```